extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -108,6 +108,11 @@ public JpaModelBuildItem discoverModelAndRegisterForReflection() throws BuildExc
         managedClassNames.addAll(collector.modelTypes);
         for (String className : managedClassNames) {
             reflectiveClass.produce(ReflectiveClassBuildItem.builder(className).methods().fields().build());
+            // Register static metamodel classes as well, so that their `class_` attribute can be populated.
+            // See org.hibernate.metamodel.internal.MetadataContext.populateStaticMetamodel
+            // Note: registering classes that do not exist is not a problem -- and is necessary if the application
+            //       tries to access these classes via reflection anyway (which it will, through Hibernate ORM).
+            reflectiveClass.produce(ReflectiveClassBuildItem.builder(className + "_").fields().build());
```

Comment:
We are positive these classes will exist?

Reply:
On the contrary, we're positive they won't exist most of the time.

But Hibernate ORM tries to load them to see if they exist, and @zakkak 's changes require that we register unexisting classes too, if an attempt to load them is done.

Alternatively we could patch Hibernate ORM to catch and ignore whatever weird exception GraalVM will throw upon attempting to load classes that are not registered for reflection. But frankly, that seems like tainting the ORM codebase with things it should not have to worry about.